corehq/apps/accounting/async_handlers.py

```diff
@@ -212,9 +213,29 @@ def account_response(self):
     @property
     def plan_version_response(self):
         edition = self.data.get('additionalData[edition]')
-        plan_versions = SoftwarePlanVersion.objects.filter(
-            plan__edition=edition
-        )
+        visibility = self.data.get('additionalData[visibility]')
+        most_recent_version = self.data.get('additionalData[most_recent_version]')
+        if most_recent_version == 'True':
```

Comment:
the body of this `if` statement is complicated enough that it might benefit from being its own function, like `get_most_recent_versions`. That would also make testing easier, as we could specifically test the functionality of that function, without having to create a web request etc.

That said, I agree with Amit. Especially given the permutations of inputs, I'd like to see some tests here, such as returning the most recent version when multiple versions are in the database, etc.

Reply:
Hi Matt, I make `get_most_recent_versions` its own function in 6e4bbdb , and write test in 6d67c8f